### PR TITLE
Pr/jais/organize completion

### PR DESCRIPTION
**Pull request by Jakobis**
Collection of lemmas and hypotheses has now been separated from Json formatting, in completionSuggester.ml.

It is worth noting that cogtopGetCompletionItems no longer deals with CompletionItems directly, but only tuples of strings

### CODE REVIEW

File: language-server/dm/completionSuggester.ml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+let getCompletionItems ~id params st loc =
+  let open Yojson.Basic.Util in
+  let hypotheses =
+    DocumentManager.get_proof st loc
+    |> Option.map (fun Proof.{ goals; sigma; _ } -> Option.cata (mk_hyps sigma) [] (List.nth_opt goals 0)) in
+  let lemmasOption : CompletionItem.completion_item list option = DocumentManager.get_lemmas st loc in
+  let lemmas = lemmasOption |> Option.map (List.map CompletionItem.pp_completion_item) in
+  [hypotheses; lemmas] 
+  |> List.map (Option.default [])
+  |> List.flatten
```
Comment (Duckapple) [inline language-server/dm/completionSuggester.ml:36]: They say `bind` when they mean `flatMap` in OCaml, apparently.
```suggestion
  |> List.bind (Option.default [])
```

File: language-server/vscoqtop/lspManager.ml [hunk 2]
```diff
@@ -377,28 +354,28 @@ let coqtopCheck ~id params =
...
-      let e, info = Exninfo.capture e in
-      let code = Lsp.LspData.Error.requestFailed in
-      let message = Pp.string_of_ppcmds @@ CErrors.iprint (e, info) in
-      output_json @@ mk_error_response ~id ~code ~message; []
-
-  let coqtopSearchResult ~id name statement =
-    let event = "vscoq/searchResult" in
-    let params = `Assoc [ "id", `String id; "name", `String name; "statement", `String statement ] in
-    output_json @@ mk_notification ~event ~params
+let coqtopSearch ~id params =
```
Comment (Hjaltesorgenfrei) [inline language-server/vscoqtop/lspManager.ml]: We should not change indentation without a reason. It just creates merge conflicts when we try to merge with the main repo

**Review by Hjaltesorgenfrei — CHANGES_REQUESTED** [review]

File: language-server/dm/completionSuggester.ml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  let (_env, hyps) =
+    Context.Compacted.fold (mk_hyp sigma)
+      (Termops.compact_named_context (Environ.named_context env)) ~init:(min_env,[]) in
+  hyps
+ 
+let getCompletionItems ~id params st loc =
+  let open Yojson.Basic.Util in
+  let hypotheses =
+    DocumentManager.get_proof st loc
+    |> Option.map (fun Proof.{ goals; sigma; _ } -> Option.cata (mk_hyps sigma) [] (List.nth_opt goals 0)) in
```
Comment (Hjaltesorgenfrei) [inline language-server/dm/completionSuggester.ml]: This should happen in mk_hyps (which should needs a new names, it gets them not just makes names for them). 
Hypotheses and lemma are two different abstraction levels hwere, which is confusing.

File: language-server/dm/completionSuggester.ml (hunk 1 above)
Comment (Jakobis) [reply language-server/dm/completionSuggester.ml:36]: Would love to, but neither bind nor flatmap exists in the list module

**Pull request merged**